### PR TITLE
list locations

### PR DESCRIPTION
**Pull request by wgresshoff**
In order to set a default location I need to know which are existing.

Closes https://github.com/inveniosoftware/invenio-files-rest/issues/234
Closes https://github.com/inveniosoftware/invenio-files-rest/issues/232

### CODE REVIEW

**Comment by ppanero** [comment]
As discussed IRL, the "set default" functionality will be generic so the create location will also make use of it when calling with ``--default``

File: invenio_files_rest/cli.py [hunk 1]
```diff
@@ -67,4 +67,33 @@ def location(name, uri, default):
...
+    click.secho('-' * 169)
+    for location in locations:
+        click.secho(str(location), fg='green')
+
+@files.command()
+@click.argument('name')
+@with_appcontext
+def default_location(name):
+    """Set default location."""
+    from .models import Location
```
Comment (ppanero) [inline invenio_files_rest/cli.py]: Imports should be on the beginning on the file.

**Comment by wgresshoff** [comment]
Just added fix for https://github.com/inveniosoftware/invenio-files-rest/issues/232

**Comment by ppanero** [comment]
Slight refactoring to fixt tests. (@wgresshoff sorry I had to take out the pretty print it is a nice addition but it requires a bigger refactor along invenio).


**Warning** it is not backwards compatible:

- CLI `location` command is now a group to contain its related commands `create`, `list` and `set_default`.
- `Location.get_by_name()` returns now `one_or_none` instead of `one` and raising an exception if it does not exist.

**Comment by ppanero** [comment]
Continued in https://github.com/inveniosoftware/invenio-files-rest/pull/236

**Pull request closed without merging**